### PR TITLE
refactored xtext gmf/jface utils to ease extensibility (no semantic change)

### PR DESCRIPTION
**Pull request by enikao**
This pull request just refactors the code in **org.yakindu.base.xtext.utils.gmf** and **org.yakindu.base.xtext.utils.jface** for enhanced extensibility:
* Introduces / extends creator methods
* Introduces protected getters for private fields
* Changes access to private fields to use getters
* Some fixed typos

This should not introduce any semantic change.

### CODE REVIEW

**Comment by andreasmuelder** [comment]
Thanks for your PR, I will have a closer look mid of next week!

**Comment by enikao** [comment]
I just added another commit that contains a semantic change, namely support for URIEditorInput in ActiveEditorTracker. This adds support for Sirius editors (without introducing any new dependencies).

File: plugins/org.yakindu.base.xtext.utils.jface/src/org/yakindu/base/xtext/utils/jface/viewers/util/ActiveEditorTracker.java [hunk 1]
```diff
@@ -150,6 +153,15 @@ public static IProject getLastActiveEditorProject() {
 		if (editorInput instanceof IFileEditorInput) {
 			final IFileEditorInput input = (IFileEditorInput) editorInput;
 			return input.getFile().getProject();
+		} else if (editorInput instanceof URIEditorInput) {
+			final URI uri = ((URIEditorInput) editorInput).getURI();
+			if (uri.isPlatformResource()) {
+				final String segment = uri.segment(1);
```
Comment (andreasmuelder) [inline plugins/org.yakindu.base.xtext.utils.jface/src/org/yakindu/base/xtext/utils/jface/viewers/util/ActiveEditorTracker.java]: Not sure if segment(1) is always the project name. This is how we map from an URI to an IFile in other places

if(uri.isPlatformResource()) {
	String platformString = uri.toPlatformString(true);
	return ResourcesPlugin.getWorkspace().getRoot().findMember(platformString).getProject();
}

**Comment by enikao** [comment]
I agree, new commit contains your proposed change.

**Comment by enikao** [comment]
Thanks for the merge (-:

**Comment by andreasmuelder** [comment]
You are welcome. 
What are you doing with the Xtext integration into sirius? =]

**Pull request merged**